xtokens/src/lib.rs

```diff
@@ -813,3 +920,18 @@ fn half(asset: &MultiAsset) -> MultiAsset {
 		id: asset.id.clone(),
 	}
 }
+
+fn subtract_fee(asset: &MultiAsset, amount: u128) -> MultiAsset {
+	let final_amount = fungible_amount(asset).checked_sub(amount).expect("fee too low; qed");
```

Comment:
There is no `fee_amount >= min_xcm_fee` check. This could panic no?

Reply:
we checked before subtract, so it's ok here I think
```
// min xcm fee should less than user fee
let fee_to_dest: MultiAsset = (fee.id.clone(), min_xcm_fee).into();
ensure!(fee_to_dest < fee, Error::<T>::InvalidAsset);
```

Reply:
maybe we need add another Error type such as `FeeNotEnough`